enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,307 @@
+    - Add decoupling ConfigMap / CustomResource:
+        Controller logic is strongly dependant of *manifests.Config struct.
+        
+
+### Example configuration
+
+
+#### current ConfigMap
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    telemeterClient:
+      enabled: false
+    prometheusK8s:
+      volumeClaimTemplate:
+        metadata:
+          name: prometheus-data
+          annotations:
+            openshift.io/cluster-monitoring-drop-pvc: "yes"
+        spec:
+          resources:
+            requests:
+              storage: 20Gi
+```
+
+### CRD
+
+```
+apiVersion: clustermonitoring.config.openshift.io
+kind: ClusterMonitoring
+metadata:
+  name: cluster
+  namespace: openshift-monitoring
+spec:
+  prometheusK8s:
+    volumeClaimTemplate:
+      metadata:
+        name: prometheus-data
+        annotations:
+          openshift.io/cluster-monitoring-drop-pvc: "yes"
+      spec:
+        resources:
+          requests:
+            storage: 20G
+```
+
+
+### Workflow Description
+
+See example in the [Proposal](#Proposal) for details.
+
+### API Extensions
+
+Add two new CRD's:
+- ClusterMonitoring
+- UserWorkloadClusterMonitoring
+
+### Topology Considerations
+
+Will function identically in all topologies.
+
+#### Hypershift / Hosted Control Planes
+No special considerations.
+
+#### Standalone Clusters
+No special considerations.
+
+#### Single-node Deployments or MicroShift
+No special considerations.
+
+### Implementation Details/Notes/Constraints
+
+
+### Migration path
+
+Migration path it will be split in three phases
+
+- Phase 1: Pre-GA (Gradual Migration for Developers)
+  - In this phase only developers are expected to use CRD.
+  - Both ConfigMap and CRD will coexist for a while until the transition is complete.
+  - Work with a Feature flag.
+  - Two options: ConfigMap or CRD.
+      Preference is to use ConfigMap over CRD during this phase.
+  - Behavior: If an entry exists in the ConfigMap, ignore the CRD.
+      If there’s no entry in the ConfigMap, use the CRD.
+  - Merge Process (only for phase 1)
+      Top-level fields (e.g., AlertmanagerMainConfig *AlertmanagerMainConfig || spec.AlertmanagerMainConfig):
+      If a field is nil in the ConfigMap, use the CRD; otherwise, stick with the ConfigMap.
+
+- Phase 2: Transition (User Control)
+  Discuss whether to keep dual support for a couple of releases.
+  - New Logic:
+    Allow users to choose the source of configuration (ConfigMap or CRD, warn about CRD preference).
+    If a CRD exists, prioritize it over the ConfigMap.
+    Add logic for users to manually switch to CRD if desired.
+    No merge process.
+
+- Phase 3: Full Migration (CRD as Default)
+    After a few releases, the migration from ConfigMap to CRD will be complete.
+    ConfigMap will be deprecated, with CRD becoming the default configuration source.
```

Comment:
In this mode, the configmap is no longer read from at all? Is that correct?

I think technically you need deprecation to be earlier than the removal, so deprecation would likely be phase 2, removal in phase 3

Reply:
If is in deprecation mode, it can be read but not recommended to do it.
Our idea is to do deprecation if phase 3, but we can also do it in phase 2. Something like in phase 2 we want to try it but not force user to do it. That is the reason we are not deprecation yet at phase 2.

In fact if everything go OK at some point phase 2 and phase 3 could be merge in just one phase - deprecation - removed.

Reply:
At some state we will remove fallback to configmap.

Reply:
You need to ensure that every user has migrated from the configmap to the CR at the point you do this, have you thought at all about how you'll ensure this?

Some things you might want to consider:
* Automatic translation to the new format
* A Cluster Fleet Evaluation that allows you to inspect which clusters have the configmap but not the CR
* An upgrade block that prevents users from upgrading until they remove the configmap (this requires staff eng approval)

Reply:
We'll can ensure that with the deprecation procedure. This question is also in open question. 

- How do we tell cluster admins that they need to migrate? CMO conditions could be an option as well as alerting and Insights?- How do we know how many clusters have migrated to the new CRD? Telemetry?

We can choose different possibilities and different ways to do it, after the development (and the issue we will find in the process) and depends and how difficult the migration will be we can think more about that and be more specific.

Reply:
> How do we tell cluster admins that they need to migrate? 

Documentation and release notes are the first step

> CMO conditions could be an option as well as alerting and Insights?

Eventually you can mark CMO as `upgradeable=false` if the configmap exists

> How do we know how many clusters have migrated to the new CRD? Telemetry?

Use [Cluster Fleet Evaluation](https://github.com/openshift/enhancements/blob/7d163511f45e73e707f8ec17f34a446772b2cff6/dev-guide/cluster-fleet-evaluation.md)

Reply:
Add info to PR